### PR TITLE
test files without warning use -Werror flag

### PR DESCRIPTION
**Pull request by cvvergara**
# Issue

Very few test files have warnings.
To avoid having in the future more files with warnings when contributors change them
This PR adds a couple of lists: tests_with_warnings
And all the tests not on that list will have the `-Werror` flag

## Tasklist

- [x] add files with warning to the lists on test and test/gurka
- [x] modify cmake to add the `-Werror` when compiling files **not** on those lists

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

**Comment by kevinkreiser** [comment]
this looks good to me but ci is complaining

**Comment by cvvergara** [comment]
Yes, sorry forgot to put the draft thing.
Was expecting to fail, haven't finished yet

**Comment by kevinkreiser** [comment]
oh I see you mean code changes for warnings to come. sounds good

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -66,19 +66,56 @@ if(NOT APPLE)
...
-    openlr.cc
-    incidents.cc
-    incident_loading.cc
-    PROPERTIES COMPILE_FLAGS "-Wall -Werror")
-endif()
+# tests files that have warning,
+# any other file not int this list will be processed with the
+# "-Werror" flag
+# Avoid adding a test to this list
+set(tests_with_warnings
```
Comment (cvvergara) [inline test/CMakeLists.txt:73]: varaible with lower cas because on gurka is used with uppercase

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -66,19 +66,56 @@ if(NOT APPLE)
...
-if (UNIX AND ENABLE_SINGLE_FILES_WERROR)
-  # Enables stricter compiler checks on a file-by-file basis
-  # which allows us to migrate piecemeal
-  set_source_files_properties(
-    openlr.cc
-    incidents.cc
-    incident_loading.cc
-    PROPERTIES COMPILE_FLAGS "-Wall -Werror")
-endif()
+# tests files that have warning,
```
Comment (cvvergara) [inline test/CMakeLists.txt:69]: With time this list should decrease in size, meaning that the file no longer has warnings
when the list is empty then all test files do not have warnings.

**Review by nilsnolde — APPROVED** [review]
Thanks, that's much less CMake code than I anticipated:)

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**